Framework/API/test/FilePropertyTest.h

```diff
@@ -184,6 +185,25 @@ class FilePropertyTest : public CxxTest::TestSuite {
     dir.remove(); // clean up your folder
   }
 
+  void testExpandUserVariables() {
+    FileProperty fp("Dir", "", FileProperty::Directory);
+    std::string msg = fp.setValue("~");
+
+    char *homepath = std::getenv("HOME");
+    if (!homepath) {
+      homepath = std::getenv("USERPROFILE");
+    }
+
+    if (homepath && Poco::File(homepath).exists()) {
```

Comment:
I don't think so, as I think `getenv` returns a null pointer if the variable is not set, so the first condition of the `&&` will fail